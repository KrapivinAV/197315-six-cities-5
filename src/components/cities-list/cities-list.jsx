import React from "react";
import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import PropTypesSet from "../../prop-types-set";
import {connect} from "react-redux";
import {changeCity, getCurrentCityOffers} from "../../store/actions";
import {cities} from "../../const";

const CitiesList = ({offers, currentCity, onCityLinkClick}) => {

  const handleCityLinkClick = (evt) => {
    evt.preventDefault();

    if (evt.currentTarget.id !== currentCity) {
      onCityLinkClick(evt.currentTarget.id, offers);
    }
  };


  return (
    <ul className="locations__list tabs__list">

      {cities.map((city) => (
        <li key={city} className="locations__item">
          <Link
            to="/"
            className={
              currentCity === city ?
                `locations__item-link tabs__item tabs__item--active` :
                `locations__item-link tabs__item`
            }
            id={city}
            onClick={handleCityLinkClick}
          >
            <span>{city}</span>
          </Link>
        </li>
      ))}

    </ul>
  );
};

CitiesList.propTypes = {
  offers: PropTypes.arrayOf(PropTypesSet.offer).isRequired,
  currentCity: PropTypes.string.isRequired,
  onCityLinkClick: PropTypes.func.isRequired,
};

const mapStateToProps = ({DATA, STATE}) => ({
  offers: DATA.offers,
  currentCity: STATE.currentCity
});

const mapDispatchToProps = (dispatch) => ({
  onCityLinkClick(id, offers) {
    dispatch(changeCity(id));
    dispatch(getCurrentCityOffers(id, offers));
  },
});

export {CitiesList};
export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
