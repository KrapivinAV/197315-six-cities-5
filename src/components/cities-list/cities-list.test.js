import React from "react";
import renderer from "react-test-renderer";
import {CitiesList} from "./cities-list";
import {Provider} from "react-redux";
import {Route, BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";

const CURRENT_CITY = `Paris`;

describe(`Render CitiesList`, () => {
  it(`Render CitiesList`, () => {

    const mockStore = configureStore([]);
    const store = mockStore({});

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Route>
                <CitiesList
                  currentCity={CURRENT_CITY}
                  onCityLinkClick={() => {}}
                />
              </Route>
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
