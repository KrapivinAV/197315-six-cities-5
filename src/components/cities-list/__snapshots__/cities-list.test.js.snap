// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CitiesList Render CitiesList 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="/"
      id="Paris"
      onClick={[Function]}
    >
      <span>
        Paris
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="/"
      id="Cologne"
      onClick={[Function]}
    >
      <span>
        Cologne
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="/"
      id="Brussels"
      onClick={[Function]}
    >
      <span>
        Brussels
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="/"
      id="Amsterdam"
      onClick={[Function]}
    >
      <span>
        Amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="/"
      id="Hamburg"
      onClick={[Function]}
    >
      <span>
        Hamburg
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="/"
      id="Dusseldorf"
      onClick={[Function]}
    >
      <span>
        Dusseldorf
      </span>
    </a>
  </li>
</ul>
`;
